core/templates/pages/exploration-editor-page/services/exploration-objective.service.ts

```diff
@@ -17,26 +17,42 @@
  * that it can be displayed and edited in multiple places in the UI.
  */
 
-require('filters/string-utility-filters/normalize-whitespace.filter.ts');
-require(
-  'pages/exploration-editor-page/services/exploration-property.service.ts');
-require('pages/exploration-editor-page/services/exploration-rights.service.ts');
-require('services/validators.service.ts');
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { ExplorationPropertyService } from './exploration-property.service';
+import { AlertsService } from 'services/alerts.service';
+import { ChangeListService } from './change-list.service';
+import { LoggerService } from 'services/contextual/logger.service';
+import { ExplorationRightsService } from './exploration-rights.service';
+import { ValidatorsService } from 'services/validators.service';
+import { NormalizeWhitespacePipe } from 'filters/string-utility-filters/normalize-whitespace.pipe';
 
-angular.module('oppia').factory('ExplorationObjectiveService', [
-  '$filter', 'ExplorationPropertyService', 'ExplorationRightsService',
-  'ValidatorsService',
-  function(
-      $filter, ExplorationPropertyService, ExplorationRightsService,
-      ValidatorsService) {
-    var child = Object.create(ExplorationPropertyService);
-    child.propertyName = 'objective';
-    child._normalize = $filter('normalizeWhitespace');
-    child._isValid = function(value) {
-      return (
-        ExplorationRightsService.isPrivate() ||
-        ValidatorsService.isNonempty(value, false));
-    };
-    return child;
+@Injectable({
+  providedIn: 'root'
+})
+export class ExplorationObjectiveService extends ExplorationPropertyService {
+  propertyName: string = 'objective';
+  constructor(
+    protected alertsService: AlertsService,
+    protected changeListService: ChangeListService,
+    protected loggerService: LoggerService,
```

Comment:
Is there any reason for making these protected instead of private?
Can you also sort these?

Reply:
yes, there is the reason for making ` alertsService,  changeListService, loggerService` protected because  ExplorationPropertyService requires these variables as protected

Reply:
Done sort